backends/p4test/midend.cpp

```diff
@@ -103,6 +104,7 @@ MidEnd::MidEnd(CompilerOptions& options, std::ostream* outStream) {
         new P4::RemoveSelectBooleans(&refMap, &typeMap),
         new P4::FlattenHeaders(&refMap, &typeMap),
         new P4::FlattenInterfaceStructs(&refMap, &typeMap),
+        new P4::FlattenHeaderUnion(&refMap, &typeMap),
```

Comment:
which pass fails? we can just remove the Predication pass if it's the one

Reply:
After invoking typedef elimination before header union flattening, all tests are passing on my local machine

Reply:
> which pass fails? we can just remove the Predication pass if it's the one

Removing Predication  pass the tests which are added to p4test's XFAIL list due to conditional statement within action.  Regressions also pass if reference output files are updated. 

Are you suggesting we can safely remove Predication pass? Or should we do in another PR to avoid any side-effect that it might cause.

Reply:
yes, we can remove it. this PR or another one isn't much of a difference really
we could also add a flag so it's applied only to controls. or perhaps a flag so that it only gives a warning and an error...